library/std/src/sys/unix/fs.rs

```diff
@@ -1180,11 +1180,13 @@ pub fn copy(from: &Path, to: &Path) -> io::Result<u64> {
                         if os_err == libc::ENOSYS
                             || os_err == libc::EXDEV
                             || os_err == libc::EINVAL
-                            || os_err == libc::EPERM =>
+                            || os_err == libc::EPERM
+                            || os_err == libc::EOPNOTSUPP =>
                     {
                         // Try fallback io::copy if either:
                         // - Kernel version is < 4.5 (ENOSYS)
                         // - Files are mounted on different fs (EXDEV)
+                        // - copy_file_range is broken in various ways on RHEL/CentOS 7 (EOPNOTSUPP)
```

Comment:
This comment could be more specific, but the exact platforms this occurs on and what's happening in the kernel is still somewhat mysterious (to me), so I think it's acceptable.

Reply:
There's some additional information in the commit message.